tests/cli/test_rasa_export.py

```diff
@@ -0,0 +1,412 @@
+import random
+
+import argparse
+import pytest
+import uuid
+from _pytest.monkeypatch import MonkeyPatch
+from _pytest.pytester import RunResult
+from pathlib import Path
+from typing import Callable, Optional, Dict, Any, Text, List
+from unittest.mock import Mock, patch
+
+import rasa.utils.io as io_utils
+from rasa.cli import export
+from rasa.core.brokers.pika import PikaEventBroker
+from rasa.core.events import UserUttered
+from rasa.core.tracker_store import TrackerStore
+from rasa.core.trackers import DialogueStateTracker
+
+
+def test_export_help(run: Callable[..., RunResult]):
+    output = run("export", "--help")
+
+    help_text = """usage: rasa export [-h] [-v] [-vv] [--quiet] [--endpoints ENDPOINTS]
+                   [--minimum-timestamp MINIMUM_TIMESTAMP]
+                   [--maximum-timestamp MAXIMUM_TIMESTAMP]
+                   [--conversation-ids CONVERSATION_IDS]"""
+
+    lines = help_text.split("\n")
+
+    for i, line in enumerate(lines):
+        assert output.outlines[i] == line
+
+
+@pytest.mark.parametrize(
+    "minimum_timestamp,maximum_timestamp",
+    [(2, 3), (None, 5.5), (None, None), (5, None)],
+)
+def test_timestamps(
+    minimum_timestamp: Optional[float], maximum_timestamp: Optional[float],
+):
+    args = argparse.Namespace()
+    args.minimum_timestamp = (
+        str(minimum_timestamp) if minimum_timestamp is not None else None
+    )
+    args.maximum_timestamp = (
+        str(maximum_timestamp) if maximum_timestamp is not None else None
+    )
+
+    # no error is raised
+    # noinspection PyProtectedMember
+    export._inspect_timestamp_options(args)
+
+
+@pytest.mark.parametrize(
+    "minimum_timestamp,maximum_timestamp",
+    [(3, 2), ("not-a-float", 5.5), (None, "no-float")],
+)
+def test_timestamp_error_exit(
+    minimum_timestamp: Optional[float], maximum_timestamp: Optional[float],
+):
+    args = argparse.Namespace()
+    args.minimum_timestamp = (
+        str(minimum_timestamp) if minimum_timestamp is not None else None
+    )
+    args.maximum_timestamp = (
+        str(maximum_timestamp) if maximum_timestamp is not None else None
+    )
+
+    with pytest.raises(SystemExit):
+        # noinspection PyProtectedMember
+        export._inspect_timestamp_options(args)
+
+
+def _write_endpoint_config_to_yaml(path: Path, data: Dict[Text, Any]) -> Path:
+    endpoints_path = path / "endpoints.yml"
+
+    # write endpoints config to file
+    io_utils.write_yaml_file(
+        data, endpoints_path,
+    )
+    return endpoints_path
+
+
+def test_get_available_endpoints(tmp_path: Path):
+    # write valid config to file
+    endpoints_path = _write_endpoint_config_to_yaml(
+        tmp_path, {"event_broker": {"type": "pika"}, "tracker_store": {"type": "sql"}}
+    )
+
+    # noinspection PyProtectedMember
+    available_endpoints = export._get_available_endpoints(str(endpoints_path))
```

Comment:
the code of `_get_available_endpoints` is repeated across multiple `cli` modules`. Let's put in a function and test it once. Also I think this `read_endpoints` should be tested as part of the `rasa.core.utils` tests and not in here.

Reply:
this is different because we want to exit from it. using an existing method and still exiting is exactly what's happening in `export._get_available_endpoints()`

Reply:
Mhm, but we are not exiting in `_get_available_endpoints`, right? So we could actually share the code of `__get_available_endpoints` . The exiting is handled later in the code.

Reply:
true, I was thinking of `get_tracker_store` and `get_event_broker` 👍

Reply:
done